var/spack/repos/builtin/packages/py-basemap/package.py

```diff
@@ -37,11 +37,23 @@ class PyBasemap(PythonPackage):
     # Per Github issue #3813, setuptools is required at runtime in order
     # to make mpl_toolkits a namespace package that can span multiple
     # directories (i.e., matplotlib and basemap)
-    depends_on('py-setuptools', type=('build', 'run'))
+    depends_on('py-setuptools', type='run')
     depends_on('py-numpy', type=('build', 'run'))
     depends_on('py-matplotlib', type=('build', 'run'))
     depends_on('pil', type=('build', 'run'))
     depends_on('geos')
 
+    def install(self, spec, prefix):
+        # The default `install` method adds `--single-version-external-managed`
+        # because the package depends on py-setuptools. But the package doesn't
+        # actually build with `py-setuptools`, it just needs it at run-time.
+        # Until I can figure out how to query direct build dependencies only,
+        # override the default `install` method.
```

Comment:
@tgamblin I mean in Spack world, not Python world. If you look at the `install()` method in `PythonPackage`, we add this flag for:

1. packages named `py-setuptools`, or
2. packages with a direct dependency on `py-setuptools`

But in the case of `py-basemap`, `py-setuptools` is a run-dependency only, not a build dependency. How do we check if its a **direct build** dependency, not just a direct dependency?